examples/analog_out/cont_gen_voltage_wfm_int_clk_every_n_samples_event.py

```diff
@@ -0,0 +1,86 @@
+"""Example of analog output voltage generation with events.
+
+This example demonstrates how to use a Every N Samples events to output a
+continuous periodic waveform to an Analog Output Channel
+using an internal sample clock. The Every N Samples events indicate when
+the specified number of samples generation is complete.
+"""
+
+from typing import Tuple
+
+import numpy as np
+import numpy.typing
+
+import nidaqmx
+from nidaqmx.constants import AcquisitionType
+
+
+def generate_sine_wave(
+    frequency: float,
+    amplitude: float,
+    sampling_rate: float,
+    phase_in: float,
+    number_of_samples: int,
+) -> Tuple[numpy.typing.NDArray[numpy.double], float]:
+    """Generates a sine wave with a specified phase.
+
+    Args:
+        frequency (float): Specifies the frequency of the sine wave.
+        amplitude (float): Specifies the amplitude of the sine wave.
+        sampling_rate (float): Specifies the sampling rate of the sine wave.
+        phase_in (float): Specifies the phase of the sine wave in radians.
+        number_of_samples (int): Specifies the number of samples to generate.
+
+    Returns:
+        Tuple[numpy.typing.NDArray[numpy.double], float]: Indicates a tuple
+        containing the generated data and the phase of the sine wave after generation.
+    """
+    duration_time = number_of_samples / sampling_rate
+    duration_radians = duration_time * 2 * np.pi
+    phase_out = (phase_in + duration_radians) % (2 * np.pi)
+    t = np.linspace(phase_in, phase_in + duration_radians, number_of_samples, endpoint=False)
+
+    return (amplitude * np.sin(frequency * t), phase_out)
+
+
+def main():
+    """Continuously generates a sine wave using an Every N Samples event."""
+    total_write = 0
+    with nidaqmx.Task() as task:
+        sampling_rate = 1000.0
+        number_of_samples = 1000
+        phase = 0.0
+
+        def callback(task_handle, every_n_samples_event_type, number_of_samples, callback_data):
+            """Callback function for Transferred N samples."""
+            nonlocal total_write
+            total_write += number_of_samples
+            print(f"Transferred data: {number_of_samples} samples. Total {total_write}.", end="\r")
+
+            return 0
+
+        task.ao_channels.add_ao_voltage_chan("Dev1/ao0")
+        task.timing.cfg_samp_clk_timing(sampling_rate, sample_mode=AcquisitionType.CONTINUOUS)
+        task.register_every_n_samples_transferred_from_buffer_event(1000, callback)
+
+        actual_sampling_rate = task.timing.samp_clk_rate
+        print(f"Actual sampling rate: {actual_sampling_rate:g} S/s")
+
+        data, phase = generate_sine_wave(
+            frequency=10.0,
+            amplitude=1.0,
+            sampling_rate=actual_sampling_rate,
+            phase_in=phase,
+            number_of_samples=number_of_samples,
+        )
```

Comment:
This example doesn't care about phase.

```suggestion
 
        def callback(task_handle, every_n_samples_event_type, number_of_samples, callback_data):
            """Callback function for Transferred N samples."""
            nonlocal total_write
            total_write += number_of_samples
            print(f"Transferred data: {number_of_samples} samples. Total {total_write}.", end="\r")

            return 0

        task.ao_channels.add_ao_voltage_chan("Dev1/ao0")
        task.timing.cfg_samp_clk_timing(sampling_rate, sample_mode=AcquisitionType.CONTINUOUS)
        task.register_every_n_samples_transferred_from_buffer_event(1000, callback)

        actual_sampling_rate = task.timing.samp_clk_rate
        print(f"Actual sampling rate: {actual_sampling_rate:g} S/s")

        data, _ = generate_sine_wave(
            frequency=10.0,
            amplitude=1.0,
            sampling_rate=actual_sampling_rate,
            number_of_samples=number_of_samples,
        )
```

Reply:
The example is updated as per suggestion.